pkg/workflow/engine_test.go

```diff
@@ -167,3 +167,30 @@ func Test_Engine_SetterRuntimeInfo(t *testing.T) {
 
 	assert.Equal(t, ri, engine.GetRuntimeInfo())
 }
+
+func Test_Engine_ClonedNetworkAccess(t *testing.T) {
+	valueName := "randomValue"
+	expected := 815
+	config := configuration.NewInMemory()
+	config.Set(valueName, expected)
+
+	engine := NewWorkFlowEngine(config)
+
+	workflowId := NewWorkflowIdentifier("cmd")
+	_, err := engine.Register(workflowId, ConfigurationOptionsFromFlagset(pflag.NewFlagSet("1", pflag.ExitOnError)), func(invocation InvocationContext, input []Data) ([]Data, error) {
+		// changing the network configuration inside a callback
+		invocation.GetNetworkAccess().GetConfiguration().Set(valueName, 1)
+		return []Data{}, nil
+	})
+	assert.Nil(t, err)
```

Comment:
Consider `assert.NoError` over `assert.Nil`. The output is a little nicer. https://go.dev/play/p/9ulULYKqeqr for example.